test/system/liquidation.test.js

```diff
@@ -0,0 +1,127 @@
+
+  let tbtcToken
+  let signerBondsSwapStrategy
+  let coveragePool
+  let riskManagerV1
+  let tbtcDeposit
+
+  let bidder
+
+  before(async () => {
+    await resetFork(startingBlock)
+
+    tbtcToken = await ethers.getContractAt("IERC20", tbtcTokenAddress)
+
+    const SignerBondsSwapStrategy = await ethers.getContractFactory(
+      "SignerBondsEscrow"
+    )
+    signerBondsSwapStrategy = await SignerBondsSwapStrategy.deploy()
+    await signerBondsSwapStrategy.deployed()
+
+    const Auction = await ethers.getContractFactory("Auction")
+
+    const masterAuction = await Auction.deploy()
+    await masterAuction.deployed()
+
+    const CoveragePoolStub = await ethers.getContractFactory("CoveragePoolStub")
+    coveragePool = await CoveragePoolStub.deploy()
+    await coveragePool.deployed()
+
+    const RiskManagerV1 = await ethers.getContractFactory("RiskManagerV1")
+    riskManagerV1 = await RiskManagerV1.deploy(
+      tbtcToken.address,
+      coveragePool.address,
+      signerBondsSwapStrategy.address,
+      masterAuction.address,
+      auctionLength
+    )
+    await riskManagerV1.deployed()
+
+    tbtcDeposit = await ethers.getContractAt("IDepositStub", depositAddress)
+
+    bidder = await impersonateAccount(bidderAddress)
+  })
+
+  describe("when auction has been fully filled", () => {
+    let auction
+    let tx
+    let bidderInitialBalance
+
+    before(async () => {
+      await tbtcDeposit.notifyRedemptionSignatureTimedOut()
+      await riskManagerV1.notifyLiquidation(tbtcDeposit.address)
+
+      const auctionAddress = await riskManagerV1.depositToAuction(
+        tbtcDeposit.address
+      )
+      auction = new ethers.Contract(auctionAddress, Auction.abi, bidder)
+      await tbtcToken.connect(bidder).approve(auction.address, lotSize)
+      bidderInitialBalance = await tbtcToken.balanceOf(bidder.address)
+      tx = await auction.takeOffer(lotSize)
+    })
+
+    it("should close auction", async () => {
+      expect(await auction.isOpen()).to.be.false
+    })
+
+    it("should liquidate the deposit", async () => {
+      // Auction bidder has spend their TBTC.
+      const bidderCurrentBalance = await tbtcToken.balanceOf(bidder.address)
+      expect(bidderInitialBalance.sub(bidderCurrentBalance)).to.equal(lotSize)
+
+      // Deposit has been liquidated.
+      expect(await tbtcDeposit.currentState()).to.equal(11) // LIQUIDATED
+    })
+
+    it("should swap signer bonds", async () => {
+      // No funds should last on the risk manager contract.
+      await expect(tx).to.changeEtherBalance(riskManagerV1, 0)
+
+      // All funds should be moved to the signer bonds processor contract.
+      await expect(tx).to.changeEtherBalance(
+        signerBondsSwapStrategy,
+        bondedAmount.mul(bondedAmountPercentage).div(100)
+      )
+    })
+
+    it("should consume a reasonable amount of gas", async () => {
+      await expect(parseInt(tx.gasLimit)).to.be.lessThan(435000)
```

Comment:
That feels... high. 🙀  The estimation from https://github.com/keep-network/coverage-pools/pull/26 was that it should cost about 89301 + 125278. 
 Is this finally reduced to 215k because we call `selfdestruct` on the contract? How would that look like without the `selfdestruct`? Gas refunds for SSTORE and SELFDESTRUCT can be removed at some point: https://eips.ethereum.org/EIPS/eip-3298.

cc @dimpar

Reply:
Yep, but estimations from #26 didn't include real-world tBTC overhead. Regarding gas refund and `selfdestruct`, I made an experiment by commenting it out and... the gas used is almost the same. It seems that `delete self;` makes the real difference as without it, gas usage is around ~40k bigger.

Reply:
The latest estimation of `takeOffer()` which also closes an auction and purchases signer bonds is about `209840` https://github.com/keep-network/coverage-pools/pull/38#discussion_r635314978
When running tests in tBTC and trying to estimate `purchaseSignerBondsAtAuction` the gas usage actually varied from around `100k` to `145k` that's why put an avg. of `125k`.
Anyway, yes, `435k` looks like almost doubles the avg. gas usage for `takeOffer()` running a full cycle with purchasing bonds.

Reply:
What if we decrease the gas limit to e.g. 220k? Will the transaction pass? That 435k gas limit could be a result of not very precise estimation. But it could also be the real amount of gas burned before some data is deleted and gas is refunded. In this case, the bidder needs to have enough ETH to cover 435k gas.

Reply:
Transaction reverts with a gas limit equal to 220k. Seems it burns around 435k and then gets a gas refund due to `delete self;`.